BeVolt/BSP/Simulator/Src/BSP_CAN.c

```diff
@@ -1,41 +1,78 @@
-#include "BSP_CAN.h"
-
-/**
- * @brief   Initializes the CAN module that communicates with the rest of the electrical system.
- * @param   None
- * @return  None
- */
-void BSP_CAN_Init(void) {
-
-    // TODO: Initialize the CAN port that connects to the rest of the car.
-}
-
-/**
- * @brief   Transmits the data onto the CAN bus with the specified id
- * @param   id : Message of ID. Also indicates the priority of message. The lower the value, the higher the priority.
- * @param   data : data to be transmitted. The max is 8 bytes.
- * @param   length : num of bytes of data to be transmitted. This must be <= 8 bytes or else the rest of the message is dropped.
- * @return  0 if module was unable to transmit the data onto the CAN bus. Any other value indicates data was transmitted.
- */
-uint8_t BSP_CAN_Write(uint32_t id, uint8_t data[8], uint8_t length) {
-    
-    // TODO: Transmit the data onto the CAN bus with the specified ID.
-
-    return 0;
-}
-
-/**
- * @brief   Gets the data that was received from the CAN bus.
- * @note    Non-blocking statement
- * @pre     The data parameter must be at least 8 bytes or hardfault may occur.
- * @param   id : pointer to store id of the message that was received.
- * @param   data : pointer to store data that was received. Must be 8bytes or bigger.
- * @return  0 if nothing was received so ignore id and data that was received. Any other value indicates data was received and stored.
- */
-uint8_t BSP_CAN_Read(uint32_t *id, uint8_t *data) {
-
-    // TODO: Check if a message has been received.
-    //      If message was received, then store the id and data of the message into the pointers and return a value other than 0.
-
-    return 0;
-}
+#include "BSP_CAN.h"
+#include <stdint.h>
+#include <stdlib.h>
+
+static const char* file = "BSP/Simulator/DataGeneration/Data/CAN.csv";
+
+
+
+/**
+ * @brief   Initializes the CAN module that communicates with the rest of the electrical system.
+ * @param   None
+ * @return  None
+ */
+
+
+
+void BSP_CAN_Init(void) {
+
+    // TODO: Initialize the CAN port that connects to the rest of the car.
+    FILE* fp = fopen(file, "w"); 
+    fclose(fp);
+}
+
+
+
+/**
+ * @brief   Transmits the data onto the CAN bus with the specified id
+ * @param   id : Message of ID. Also indicates the priority of message. The lower the value, the higher the priority.
+ * @param   data : data to be transmitted. The max is 8 bytes.
+ * @param   length : num of bytes of data to be transmitted. This must be <= 8 bytes or else the rest of the message is dropped.
+ * @return  0 if module was unable to transmit the data onto the CAN bus. Any other value indicates data was transmitted.
+ */
+
+uint8_t BSP_CAN_Write(uint32_t id, uint8_t data[8], uint8_t length) {
+    uint32_t offset;
+    // TODO: Transmit the data onto the CAN bus with the specified ID.
+    
+

```

Comment:
Can you format the file with the CAN messages in a more human-readable format? Maybe something like
CAN id: 0x???
data: 0x??, 0x??, etc.
You also don't need to put the length of the CAN message, but if you want to, you could do something like this:
length: (some number)

Reply:
Since these are CSV files it makes it easier to read and write if there aren’t any row names. I feel like since the python simulator will be reading and displaying the data we shouldn’t be viewing the CSV files anyway. Just my thoughts but up to you.

Reply:
That's a good point. Maybe just separate all of the data bytes with commas?

Reply:
We will need a method in the .py file to parse the csv file if we are going to use on python to read the values